### DISCUSSION TITLE
Extract Motif PFMs for MotifMatrix features

### DISCUSSION THREAD

**Discussion by jindalk** [GreenleafLab/ArchR#1429] [General]
Hi,
Is it possible to extract Position Frequency Matrices for the 884 motifs used for the chromVAR analysis? I am trying to run Tomtom for some of the motif targets that I have identified in my dataset: https://meme-suite.org/meme/tools/tomtom

Thanks,
Kunal

**Comment by Zepeng-Mu** [comment]
Maybe this https://github.com/GreenleafLab/ArchR/discussions/1226?

**Reply by jindalk** [reply]
Ah thanks! One more question, since I need to convert these PWMs to PFMs for Tomtom, would you know what pseudocount was used to obtain these PWMs? I dug around a bit and found this in the original chromVAR paper - These position frequency matrices were then converted into position weight matrices (PWMS) by taking the log of the frequency after adding a 0.008 pseudocount and dividing by 0.25.

I wonder if the same params were used in ArchR motif models as well?

**Reply by Zepeng-Mu** [reply]
I think (I could be wrong) ArchR is using 0.008 as well, and this is also the default used in `TFBSTools` package.

**Reply by rcorces** [reply]
I also believe this is correct.

**Reply by jindalk** [reply]
Got it, thank you both!

**Comment by jindalk** [comment]
Hi,
I tried the method that was suggested to extract PFMs, but I am not sure if I'm still getting the correct matrix. This is the code I used:

`all_motifs <- b4.atac@peakAnnotation[['Motif']]$motifs`
`foxa1_motif <- TFBSTools::as.matrix(all_motifs[['Foxa1_306']])`
`foxa1_pfm <- 0.25*(2**foxa1_motif) - 0.008`

In the final matrix, when I compute colSums, it is not equal to ~1 for each of the positions. It is actually as low as 0.7 in some cases.
`colSums(foxa1_pfm)`
` 0.9484407 0.8164749 0.8840974 0.8772556 0.7316546 0.7721702 0.7592677 0.7276554 0.6784628 0.7938169
[11] 0.6837447 0.9599072 0.9553515 0.9654302 0.9576198`

Shouldn't colSums of PFM by definition be equal to 1, unless I am understanding this incorrectly?

Thanks

**Reply by rcorces** [reply]
maybe this helps https://github.com/GreenleafLab/ArchR/issues/476#issuecomment-753241731

**Reply by jindalk** [reply]
The issue linked below that comment (#509) solved it for me. I was using log 2 but it had to be natural log instead. Thank you!
